### PR TITLE
[GDBJIT] Provide full file names in generated DWARF

### PR DESCRIPTION
**Pull request by ayuckhulk**
This PR enables GDBJIT interface to provide full file names in GDBJIT-generated DWARF.

There are two main changes:
* Fix compile unit path - this should be a full path to the source file, not to the executable module
* Add Directory Table info to the line info section - this enables LLDB to display a full path for the source file

CC @Dmitri-Botcharnikov

### CODE REVIEW

**Comment by ayuckhulk** [comment]
@mikem8361 @janvorli PTAL

File: src/vm/gdbjit.cpp [hunk 1]
```diff
@@ -2893,98 +2908,205 @@ void NotifyGdb::MethodPitched(MethodDesc* methodDescPtr)
...
     }
 
-    /* build file table */
-    unsigned totalSize = 0;
+public:
 
-    for (unsigned i = 0; i < nfiles; ++i)
+    FileTableBuilder(int capacity) :
+        m_capacity(capacity),
+        m_dirs(new NewArrayHolder<char>[capacity]),
```
[the remark this answers is not in the document]
Comment (janvorli) [reply src/vm/gdbjit.cpp]: I am sorry, wrong comment.

File: src/vm/gdbjit.cpp [hunk 2]
```diff
@@ -2893,98 +2908,205 @@ void NotifyGdb::MethodPitched(MethodDesc* methodDescPtr)
...
     buf.MemPtr = new char[buf.MemSize];
 
     /* Fill the line info header */
     DwarfLineNumHeader* header = reinterpret_cast<DwarfLineNumHeader*>(buf.MemPtr.GetValue());
     memcpy(buf.MemPtr, &LineNumHeader, sizeof(DwarfLineNumHeader));
-    header->m_length = buf.MemSize - sizeof(uint32_t);
-    header->m_hdr_length = sizeof(DwarfLineNumHeader) + 1 + fileTable.MemSize - 2 * sizeof(uint32_t) - sizeof(uint16_t);
-    buf.MemPtr[sizeof(DwarfLineNumHeader)] = 0; // this is for missing directory table
+    header->m_length = buf.MemSize - sizeof(header->m_length);
+    header->m_hdr_length = sizeof(DwarfLineNumHeader)
```
[the remark this answers is not in the document]
Comment (ayuckhulk) [reply src/vm/gdbjit.cpp]: Sure. It's the number of bytes following the `m_hdr_field` field to the beginning of the first byte of
the line number program itself. See http://www.dwarfstd.org/doc/Dwarf3.pdf Page 95 (at the bottom).

[the remark this answers is not in the document]
Comment (janvorli) [reply src/vm/gdbjit.cpp]: Thank you!

**Comment by ayuckhulk** [comment]
@janvorli Thank you for the review!

**Pull request merged**